study/src/test/java/di/stage4/annotations/DIContainer.java

```diff
@@ -9,16 +16,48 @@ class DIContainer {
 
     private final Set<Object> beans;
 
-    public DIContainer(final Set<Class<?>> classes) {
-        this.beans = Set.of();
+    public DIContainer(Set<Class<?>> classes) {
+        this.beans = createBeans(classes);
+        this.beans.forEach(this::assignFields);
     }
 
     public static DIContainer createContainerForPackage(final String rootPackageName) {
-        return null;
+        Set<Class<?>> classes = ClassPathScanner.getAllClassesInPackage(rootPackageName);
+        return classes.stream()
+                .filter(DIContainer::hasServiceOrRepositoryAnnotation)
+                .collect(Collectors.collectingAndThen(Collectors.toSet(), DIContainer::new));
```

Comment:
[제안]

변경되지 말아야 하는 Set에는 불변화를 시켜주어도 좋을 것 같습니다.

```suggestion
                .collect(Collectors.collectingAndThen(Collectors.unmodifiableSet(), DIContainer::new));
```

Reply:
[반영] [d8e7861](https://github.com/woowacourse/java-mvc/pull/835/commits/d8e786158db5b9457f222f3570639db91024f59f)